### PR TITLE
Learning rate decay

### PR DESCRIPTION
**Pull request by johnpaulbin**
Adds learning rate decay to trainer/Tacotron2.py

New default hparams:

lr_decay_start=15000, # The global step to start decay
lr_decay_rate=216000, # Rate which to decay. 
Equation: `learning_rate = (self.learning_rate * (EulerNumber ** (-self.global_step / self.lr_decay_rate)))`
lr_decay_min=1e-5 # Minimum LR

### CODE REVIEW

File: uberduck_ml_dev/trainer/tacotron2.py [hunk 1]
```diff
@@ -461,6 +464,15 @@ def train(self):
                 sampler.set_epoch(epoch)
             for batch_idx, batch in enumerate(train_loader):
                 self.global_step += 1
+                
+                # Learning Rate decay, can be disabled if lr_decay_start is == 0 or None
+                if (self.global_step > self.lr_decay_start) and (self.lr_decay_start not in [0, None]):
+                    learning_rate = (self.learning_rate * (EulerNumber ** (-self.global_step / self.lr_decay_rate)))
```
Comment (sjkoelle) [inline uberduck_ml_dev/trainer/tacotron2.py:470]: use `np.exp()` or `torch.exp` instead of EulerNumber

**Review by sjkoelle — APPROVED** [review]
